finat/hdivcurl.py

```diff
@@ -202,3 +226,78 @@ def select_hcurl_transformer(element):
                           gem.Zero()]
     else:
         assert False, "Unexpected original mapping!"
+
+
+def select_inverse_hdiv_transformer(element):
+    # Assume: something x interval
+    assert len(element.factors) == 2
+    assert element.factors[1].cell.get_shape() == LINE
+
+    # Globally consistent edge orientations of the reference
+    # quadrilateral: rightward horizontally, upward vertically.
+    # Their rotation by 90 degrees anticlockwise is interpreted as the
+    # positive direction for normal vectors.
+    ks = tuple(fe.formdegree for fe in element.factors)
+    if ks == (0, 1):
```

Comment:
This function requires that the weight is a `gem.Literal`. I suspect that that might not be the case forever.

Reply:
`select_hdiv_transformer` and `select_hcurl_transformer` also require that the lambda input is a gem object.

Reply:
Sure, but they don't make assumptions about what type of gem object it is.

Reply:
Would changing the input and output to be a simple tuple or list, and letting the dual basis method handle the conversion to a `Literal` be better?